src/main/java/de/tum/cit/aet/artemis/exercise/service/ExerciseService.java

```diff
@@ -372,10 +373,11 @@ public Set<Exercise> filterExercisesForCourse(Course course, User user) {
      * Loads additional details for team exercises and for active quiz exercises
      * Assumes that the exercises are already been loaded (i.e. no proxy)
      *
-     * @param course corresponding course: exercises
-     * @param user   the user entity
+     * @param course          corresponding course: exercises
+     * @param user            the user entity
+     * @param loadQuizBatches only necessary when loading one course
```

Comment:
Same question here

Reply:
this is a temporary fix, we should completely separate the code between loading multiple courses and one course in the future. Loading the quiz batches was necessary some time ago, otherwise the exercise status was not displayed correctly. After the huge design changes, this might not be necessary anymore, in particular because the exercise is now loaded again when navigating into it.
We basically need to check what data is shown in the sidebar to determine if loading quiz batches is still relevant here or not